packages/react/src/components/button.tsx

```diff
@@ -116,6 +119,7 @@ export function UploadButton<
   const { startUpload, isUploading, routeConfig } = useUploadThing(
     $props.endpoint,
     {
+      signal: $props.signal ?? acRef.current.signal,
```

Comment:
Q: Should we omit the `signal` from the props for built-in components or allow them to pass an `AbortController` instead?

Not sure what the usage will look like but as it is now passing in a signal as prop will break the built-in cancellation mechanism. Or should we disable the "cancel on hover" behavior on the button if they pass in a signal?

I'm not sure what the right path is here... I think I'd be fine not allowing custom AbortSignal's for built-in components?

Reply:
> I think I'd be fine not allowing custom AbortSignal's for built-in components?

I am inclined to agree. If you need to do something more complex than the cancellation logic we provide, it makes sense that you build your own components.

Reply:
https://github.com/pingdotgg/uploadthing/pull/838/commits/d6ebba061480185e62ad23f464b1abf96c5477fd